### PR TITLE
examples: Add observability example

### PR DESCRIPTION
**Pull request by zasweq**
This PR adds an Observability example. It consists of the Hello World example with Observability module turned on.

RELEASE NOTES: N/A

### CODE REVIEW

File: examples/features/observability/server/main.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	flag.Parse()
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", *port))
+	if err != nil {
+		log.Fatalf("failed to listen: %v", err)
+	}
+	s := grpc.NewServer()
+	pb.RegisterGreeterServer(s, &server{})
+	log.Printf("server listening at %v", lis.Addr())
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
```
Comment (dfawley) [inline examples/features/observability/server/main.go:74]: Please add some comments about this, too.

File: examples/features/observability/go.mod [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	google.golang.org/genproto v0.0.0-20230125152338-dcaf20b6aeaa // indirect
+	google.golang.org/grpc/stats/opencensus v0.0.0-20230317183452-b638faff2204 // indirect
+	google.golang.org/protobuf v1.29.1 // indirect
+)
+
+replace google.golang.org/grpc => ../../..
+
+replace google.golang.org/grpc/gcp/observability => ../../../gcp/observability
+
+replace google.golang.org/grpc/stats/opencensus => ../../../stats/opencensus
```
Comment (dfawley) [inline examples/features/observability/go.mod:47]: I think we should remove these so the go.mod is as clean as possible (i.e. closest to what users should be using).

File: examples/features/observability/go.mod [hunk 3]
```diff
@@ -0,0 +1,47 @@
+module google.golang.org/grpc/examples/features/observability
+
+go 1.17
+
+require (
+	google.golang.org/grpc v1.54.0
+	google.golang.org/grpc/examples v0.0.0-20230323213306-0fdfd40215dc
+	google.golang.org/grpc/gcp/observability v0.0.0-00010101000000-000000000000
```
Comment (dfawley) [inline examples/features/observability/go.mod:8]: Let's not forget to circle back and make these `v1.0.0` after release.

**Review by dfawley — APPROVED** [review]

File: examples/features/observability/README.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+This example is the Hello World example instrumented for logs, metrics, and tracing.
+
+Please refer to Microservices Observability user guide for setup.
```
Comment (dfawley) [inline examples/features/observability/README.md:3]: Add a link here?

File: examples/features/observability/server/main.go (hunk 1 above)
Comment (zasweq) [reply examples/features/observability/server/main.go:74]: Done. Let me know if comment is too verbose.

File: examples/features/observability/README.md (hunk 4 above)
Comment (dfawley) [reply examples/features/observability/README.md:3]: I guess that's true.. we have a user guide but it doesn't have the instructions for running this yet.

File: examples/features/observability/go.mod (hunk 3 above)
Comment (zasweq) [reply examples/features/observability/go.mod]: Will get back to after releasing stats/opencensus and gcp/observability.

File: examples/features/observability/server/main.go (hunk 1 above)
Comment (dfawley) [reply examples/features/observability/server/main.go:81]: How about:

```
// This server can only be terminated by an external signal from the Operating System.
// The following catches those signals and initiates s.Stop().  This causes the s.Serve()
// call to return and run main()'s defers, including the observability.End() call that
// ensures any pending observability data is sent to Cloud Operations. 
```

**Review by dfawley — APPROVED** [review]

File: examples/features/observability/server/main.go (hunk 1 above)
Comment (zasweq) [reply examples/features/observability/server/main.go:81]: Actually, will say "this server can potentially" to keep the rest of comment

**Pull request merged**